frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/behaviors/assign-leader/simulated-region-overview-behavior-assign-leader.component.ts

```diff
@@ -23,13 +22,13 @@ export class SimulatedRegionOverviewBehaviorAssignLeaderComponent
 
     currentLeader?: Observable<Personnel>;
 
-    constructor(private readonly store: Store<AppState>) {}
+    constructor(private readonly storeService: StoreService) {}
 
     ngOnChanges(): void {
-        if (this.assignLeaderBehaviorState.leaderId) {
-            this.currentLeader = this.store.select(
-                createSelectPersonnel(this.assignLeaderBehaviorState.leaderId)
-            );
-        }
+        this.currentLeader = this.assignLeaderBehaviorState.leaderId
+            ? this.storeService.select$(
+                  createSelectPersonnel(this.assignLeaderBehaviorState.leaderId)
+              )
+            : undefined;
```

Comment:
I believe I fixed a potential bug here:
If the `leaderId` would have changed back from an id to `undefined`, the component would still have displayed the previous leader.